packages/xo-web/src/common/xo/index.js

```diff
@@ -655,14 +655,26 @@ export const enableHost = host => _call('host.enable', { id: resolveId(host) })
 export const disableHost = host =>
   _call('host.disable', { id: resolveId(host) })
 
-export const getHostMissingPatches = host =>
-  _call('host.listMissingPatches', { host: resolveId(host) }).then(
-    patches =>
-      // Hide paid patches to XS-free users
-      host.license_params.sku_type !== 'free'
-        ? patches
-        : filter(patches, ['paid', false])
-  )
+const missingUpdatePluginByHost = {}
+export const getHostMissingPatches = async host => {
+  const hostId = resolveId(host)
+  if (host.productBrand !== 'XCP-ng') {
+    const patches = await _call('host.listMissingPatches', { host: hostId })
+    // Hide paid patches to XS-free users
+    return host.license_params.sku_type !== 'free'
```

Comment:
Out of curiosity, was isn't this done server-side?

Reply:
I think we wanted to be able to tell the user that there were some missing patches that he wasn't able to install because of XenServer restrictions. But we haven't actually used it so far.